.github/workflows/pre-commit.yml

```diff
@@ -7,51 +7,92 @@ on:
       - master
 
 jobs:
-  getBaseVersion:
-    name: Get min/max versions
+  # Min Terraform version(s)
+  getDirectories:
+    name: Get root directories
     runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install Python
+        uses: actions/setup-python@v2
+      - name: Build matrix
+        id: matrix
+        run: |
+          DIRS=$(python -c "import json; import glob; import re; print(json.dumps([x.replace('/versions.tf', '') for x in glob.glob('./**/versions.tf', recursive=True) if not re.match(r'^.+/_', x)]))")
```

Comment:
@bryantbiggs I made a small change here to skip directories starting with an underscore (such as `modules/_templates`) because they are partial and should not be processed.